jni/include/nmslib_stream_support.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+#ifndef OPENSEARCH_KNN_JNI_NMSLIB_STREAM_SUPPORT_H
+#define OPENSEARCH_KNN_JNI_NMSLIB_STREAM_SUPPORT_H
+
+#include "jni_util.h"
+#include "native_engines_stream_support.h"
+
+#include <jni.h>
+#include <stdexcept>
+#include <iostream>
+#include <cstring>
+
+namespace knn_jni {
+namespace stream {
+
+
+
+/**
+ * TODO : KDY
+ */
+class NmslibMediatorInputStreamBuffer final : public std::streambuf {
+ public:
+  explicit NmslibMediatorInputStreamBuffer(NativeEngineIndexInputMediator *_mediator)
+      : std::streambuf(),
+        mediator(_mediator) {
+  }
+
+ protected:
+  std::streamsize xsgetn(std::streambuf::char_type *destination, std::streamsize count) final {
```

Comment:
What does this name mean? Also, why is this access protected?

Reply:
Please refer to the official doc : https://en.cppreference.com/w/cpp/io/basic_streambuf/sgetn
In short, `xsgetn` is a protect virtual method defined in `std::basic_streambuf`, which is called whenever `sgetn` is called for the read delegation to the implementation.

x -> extended
s -> sequence

xs + getn = extended sequence getter function.

Reply:
Oh this is really cool

Reply:
Please add this explanation as comments on the function. Hard to remember for future.